pkg/controller/appstatus/endpoints.go

```diff
@@ -90,7 +94,17 @@ func serviceEndpoints(ctx context.Context, c kclient.Client, app *v1.AppInstance
 		}
 	}
 
-	return
+	set := map[string]v1.Endpoint{}
+	for _, ep := range endpoints {
+		set[fmt.Sprintf("%v://%v", ep.Protocol, ep.Address)] = ep
+	}
```

Comment:
Instead of constructing a slice, then constructing a map, and then constructing a slice again, can the main for-loop above construct the map? That way we dedup as we go?

Reply:
Great call. I can do that.